src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoderBasicLatin.cs

```diff
@@ -0,0 +1,289 @@
+
+            // Forbid characters that are special in HTML.
+            // Even though this is a not HTML encoder,
+            // it's unfortunately common for developers to
+            // forget to HTML-encode a string once it has been JS-encoded,
+            // so this offers extra protection.
+            DefaultHtmlEncoder.ForbidHtmlCharacters(allowedCharacters);
+
+            // '\' (U+005C REVERSE SOLIDUS) must always be escaped in Javascript / ECMAScript / JSON.
+            // '/' (U+002F SOLIDUS) is not Javascript / ECMAScript / JSON-sensitive so doesn't need to be escaped.
+            allowedCharacters.ForbidCharacter('\\');
+
+            // '`' (U+0060 GRAVE ACCENT) is ECMAScript-sensitive (see ECMA-262).
+            allowedCharacters.ForbidCharacter('`');
+
+#if DEBUG
+            // Verify and ensure that the AllowList bit map matches the set of allowed characters using AllowedCharactersBitmap
+            for (int i = 0; i < AllowList.Length; i++)
+            {
+                char ch = (char)i;
+                Debug.Assert((allowedCharacters.IsCharacterAllowed(ch) ? 1 : 0) == AllowList[ch]);
+                Debug.Assert(allowedCharacters.IsCharacterAllowed(ch) == !NeedsEscaping(ch));
+            }
+            for (int i = AllowList.Length; i <= char.MaxValue; i++)
+            {
+                char ch = (char)i;
+                Debug.Assert(!allowedCharacters.IsCharacterAllowed(ch));
+                Debug.Assert(NeedsEscaping(ch));
+            }
+#endif
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public override bool WillEncode(int unicodeScalar)
+        {
+            if (UnicodeHelpers.IsSupplementaryCodePoint(unicodeScalar))
+            {
+                return true;
+            }
+
+            Debug.Assert(unicodeScalar >= char.MinValue && unicodeScalar <= char.MaxValue);
+
+            return NeedsEscaping((char)unicodeScalar);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public override unsafe int FindFirstCharacterToEncode(char* text, int textLength)
+        {
+            if (text == null)
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+
+            int idx = 0;
+
+#if NETCOREAPP
+            if (Sse2.IsSupported)
+            {
+                short* startingAddress = (short*)text;
+                while (textLength - 8 >= idx)
+                {
+                    Debug.Assert(startingAddress >= text && startingAddress <= (text + textLength - 8));
+
+                    // Load the next 8 characters.
+                    Vector128<short> sourceValue = Sse2.LoadVector128(startingAddress);
+
+                    // Check if any of the 8 characters need to be escaped.
+                    Vector128<short> mask = Sse2Helper.CreateEscapingMask_DefaultJavaScriptEncoderBasicLatin(sourceValue);
+
+                    int index = Sse2.MoveMask(mask.AsByte());
```

Comment:
Similar to my other comment: why is `AsByte` called (since we just initialized for `short`)… I won't comment on other cases.

Reply:
We don't have a `short` overload on `MoveMask`.